### DISCUSSION TITLE
Dinamic custom classes with weird behivor - Next.js

### DISCUSSION THREAD

**Discussion by KatuGT** [tailwindlabs/tailwindcss#11475] [Help] [answered]
Hello, will show you what is happening in this gif

https://imgur.com/YFsnwQt
(sorry, to big to be shown directly)

Well, I have created a  custom class for each size and letter shown in the table and the SVG image. The problem is on the tabla below. Yesterday it was working just fine, today insted of all the letters (A, B, C, D, E, F) were all white and ofcourse the hover change wasn´t working

weirdly the correct colors just show up if I apply to another element and they stay, even If I reload the web

I will leave here a portion of the code:

` <thead>
            <tr className="border-b border-solid border-gray-200">
              <th className="w-32 p-2">Make</th>
              <th className="w-32 p-2">Code</th>
              {sizes?.map((size) => {
                const needDataList = size?.inputName?.includes("chain");

                return (
                  <th
                    key={size?.inputName}
                    onMouseEnter={handleHover}
                    onMouseLeave={handleMouseLeave}
                    className={`size${size?.displayName} w-28 ${
                      hoverClass === `size${size?.displayName}`
                        ? `text-size${size?.displayName}colorLight`
                        : `text-size${size?.displayName}color`
                    }`}
                  >`
The classes are being applied  correctly as far I can see here
![image](https://github.com/tailwindlabs/tailwindcss/assets/38865485/b5915a2b-4309-49b5-a0d7-905191767dfa)

And what happens if I dont use my custom classes? it works fine...
![ScreenRecorderProject2](https://github.com/tailwindlabs/tailwindcss/assets/38865485/7396e132-47a0-4a9b-ae8f-da802c4a1ace)

This is the exact path to the component where the Table is used and from where the table get the info
https://github.com/KatuGT/sizematch/blob/front/frontSprocket/src/app/FrontSprocket/page.tsx

and this one the Table itself
https://github.com/KatuGT/sizematch/blob/front/frontSprocket/src/Components/Table/Table.tsx

note: the colors of the SVG are applied not from tailwind, but from a simple array

**Comment by wongjn** [comment, marked as answer]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

**Reply by KatuGT** [reply]
that´s seems exactly what I did 😥 It´s weird it worked while doing it and the next days is broken

Thankyou Wongjn! I will have too find a different approach
